packages/hardhat/contracts/strategy/CrabStrategyV2.sol

```diff
@@ -604,96 +604,89 @@ contract CrabStrategyV2 is StrategyBase, StrategyFlashSwap, ReentrancyGuard, Own
             _order.trader, // market maker
-            _order.managerAmount, // token out
-            _order.traderAmount, // token in
-            _orderPrice
+            _order.quantity, // order oSQTH quantity
+            _order.price, // order price
+            _order.isBuying, // order direction
+            _clearingPrice // executed price for order
         );
     }
 
     /**
      * @dev hedge function to reduce delta using an array of signed orders
-     * @param _managerSellAmount quantity the manager wants to sell
-     * @param _clearingPrice the price at which the manager is buying
+     * @param quantity quantity the manager wants to trade
+     * @param _clearingPrice clearing price in weth
+     * @param _isBuying direction of manager trade
      * @param _orders an array of signed order to swap tokens
      */
     function hedgeOTC(
-        uint256 _managerSellAmount,
+        uint256 _quantity,
         uint256 _clearingPrice,
+        bool _isBuying,
         Order[] memory _orders
     ) external onlyOwner {
         require(_clearingPrice > 0, "Manager Price should be greater than 0");
         require(_isTimeHedge() || _isPriceHedge(), "Time or Price is not within range");
-        _checkOTCPrice(_clearingPrice, _orders[0].managerToken);
+        _checkOTCPrice(_clearingPrice, _isBuying);
 
         timeAtLastHedge = block.timestamp;
 
-        uint256 remainingAmount = _managerSellAmount;
+        uint256 remainingAmount = _quantity;
         uint256 prevPrice = 0;
         uint256 currentPrice = 0;
 
-        bytes32 firstTradePair = keccak256(abi.encode(_orders[0].traderToken, _orders[0].managerToken));
-        bytes memory tradePair;
         for (uint256 i = 0; i < _orders.length; i++) {
-            tradePair = abi.encode(_orders[i].traderToken, _orders[i].managerToken);
-            currentPrice = _orders[i].managerAmount.mul(1e18).div(_orders[i].traderAmount);
-            require(currentPrice >= prevPrice, "Orders are not arranged properly");
-            if (i > 0) {
-                require(keccak256(tradePair) == firstTradePair, "All orders must have the same buy/sell token.");
-            }
+            currentPrice = _orders[i].price;
+            if (_isBuying){
+                require(currentPrice >= prevPrice, "Orders are not arranged properly");
+                } else {
+                require(currentPrice <= prevPrice, "Orders are not arranged properly");                    
+                }
             prevPrice = currentPrice;
 
-            if (remainingAmount > _orders[i].managerAmount) {
-                _execOrder(remainingAmount, _clearingPrice, currentPrice, _orders[i]);
-                remainingAmount = remainingAmount.sub(_orders[i].managerAmount);
-            } else {
-                _execOrder(remainingAmount, _clearingPrice, currentPrice, _orders[i]);
-                break;
-            }
+            _execOrder(remainingAmount, _clearingPrice, _orders[i]);
+
```

Comment:
we need to keep track of the remainingAmount I think, its never updated here, was removed from this function

Reply:
yep back now